efficientnetv2/main_tf2.py

```diff
@@ -118,189 +92,349 @@ def test_step(self, data):
+        if not self.config.train.stages:
+          _train_dataset = self.get_dataset(training=True, image_size=self.train_size, config=self.config)
+          _fit_config.update({'train_data': _train_dataset})
+
+        else:
+          # train model on multi stages.
+          self.train_multi_stage(_fit_config)
+
+      # training the model with corresponding to mode.
+      if not self.config.train.stages:
+        model.fit(_fit_config)
+
+    elif self.mode == 'eval':
+      self.evaluate()
+
+
+  def get_dataset(self, training: bool, image_size: Tuple[int, int, int], config: Config) -> Any:
+    """A shared utility to get input dataset."""
+    if training:
+      return self.strategy.distribute_datasets_from_function(
+          datasets.build_dataset_input(
+              True, image_size, self.image_dtype, FLAGS.data_dir, train_split,
+              config.data).distribute_dataset_fn(config.train.batch_size))
+    return self.strategy.distribute_datasets_from_function(
+        datasets.build_dataset_input(
+            False, image_size, self.image_dtype, FLAGS.data_dir, eval_split,
+            config.data).distribute_dataset_fn(config.eval.batch_size))
+
+
+  def set_precision(self) -> None:
+    """ Sets precision for the training session. """
+    self.image_dtype = 'bfloat16' if self.strategy == 'tpu' else 'float16'
+    precision = 'mixed_bfloat16' if self.strategy == 'tpu' else 'mixed_float16'
+    logging.info('Training precision set to %s', precision)
+    policy = tf.keras.mixed_precision.Policy(precision)
+    tf.keras.mixed_precision.set_global_policy(policy)
+
+
+  def get_config(self, base_config: Config) -> Config:
+    """ Get training config. """
+    config = copy.deepcopy(base_config)
+    config.override(effnetv2_configs.get_model_config(FLAGS.model_name))
+    config.override(datasets.get_dataset_config(FLAGS.dataset_cfg))
+    config.override(FLAGS.hparam_str)
+    config.model.num_classes = config.data.num_classes
+    return config
+
+  def get_strategy(self) -> None:
+    """ Get ds_strategy for the training session. """
+
+    if strategy == 'tpu':
+      tpu_cluster_resolver = tf.distribute.cluster_resolver.TPUClusterResolver(
+          FLAGS.tpu, zone=FLAGS.tpu_zone, project=FLAGS.gcp_project)
+      tf.config.experimental_connect_to_cluster(tpu_cluster_resolver)
+      tf.tpu.experimental.initialize_tpu_system(tpu_cluster_resolver)
+      ds_strategy = tf.distribute.TPUStrategy(tpu_cluster_resolver)
+      logging.info('All devices: %s', tf.config.list_logical_devices('TPU'))
+    elif strategy == 'gpus':
+      ds_strategy = tf.distribute.MirroredStrategy()
+      logging.info('All devices: %s', tf.config.list_physical_devices('GPU'))
+    else:
+      if tf.config.list_physical_devices('GPU'):
+        ds_strategy = tf.distribute.MirroredStrategy(['GPU:0'])
+      else:
+        ds_strategy = tf.distribute.MirroredStrategy(['CPU:0'])
+    return ds_strategy
+
+
+
+
+
+
+
+
+
```

Comment:
Too many empty lines?

Reply:
Sorry will remove em 
PR is still in progress